src/Ng/Jobs/Catalog2RegistrationJob.cs

```diff
@@ -118,6 +120,20 @@ protected override void Init(IDictionary<string, string> arguments, Cancellation
             TelemetryService.GlobalDimensions[TelemetryConstants.Destination] = _destination?.AbsoluteUri;
         }
 
+        private int MaxConcurrentBatches(int argBatches)
```

Comment:
IMO this logic should be in `RegistrationCollector`'s constructor instead so that no caller can pass in an invalid number of batches.

Reply:
I agree.. but other places we do it when reading arguments, I would keep it consistent.